common/dev/peci.c

```diff
@@ -0,0 +1,268 @@
+#include "peci.h"
+
+#include <drivers/peci.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "sensor.h"
+#include "hal_peci.h"
+#include "plat_sensor_table.h"
+
+#define PECI_READ_LEN 5
+#define PECI_RETRY_MAX 6
+
+#ifdef CONFIG_PECI
+static bool peci_get_power(uint8_t sensor_num, int *reading)
```

Comment:
Check reading != NULL

Reply:
The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .